### ISSUE TITLE
--net=none lacks useful DNS options

### ISSUE THREAD

**Issue by nbastin** [docker/libnetwork#1063]
None of the existing DNS options make any sense for containers where --net=none (and the embedded DNS in 1.10 makes even more of a mess where it decides to add public DNS).

Can we get something like --dns=ignore or something, where it mounts an empty resolv.conf file and doesn't get creative with trying to do some magical right thing, and never refreshes based on the host resolv.conf.

This probably isn't the right solution, but there needs to be some kind of solution.

**Comment by mavenugo** [comment]
@nbastin as discussed in the IRC, I dont think the embedded DNS server in 1.10 has nothing to do with the behavior that you see in --net=none. Also as discussed, can you please confirm which version of docker the behavior has changed (as per your requirement) ?

**Comment by nbastin** [comment]
Yeah, I'm not actually sure where this changed.  It's definitely not 1.10 and embedded DNS isn't actually the source of this problem (although I would still like it to not be around if I'm not using it), so that part of the original report isn't valid.

My scripts for installing old dockers don't appear to work anymore (they did some auto-sed magic on the download script that apparently doesn't work and I haven't debugged it), so I'm not sure when this actually changed.  However, I don't think that's particularly relevant - I think really I'm asking for a feature that never explicitly existed (not one that worked before but is now broken).

So, what I really mean here is - can I get a world where --net=none is more pervasive than it is now?  I *think* this means just not munging /etc/resolv.conf now, but also making sure that networking (and networking-adjacent filesystem changes, which is really what /etc/resolv.conf is at some level) changes moving forward have no effect on containers created with --net=none.

**Comment by mavenugo** [comment]
@nbastin can you please also edit your description accordingly ? Also it would be good to know which docker version  were you using earlier and which version brought in this changed behavior ? This will help us narrow down the root-cause and if it will have any impact on other users.

**Comment by nbastin** [comment]
I added https://github.com/docker/libnetwork/issues/1928 in order to try to encapsulate the current state of the problems that started showing up with this issue.  Since new installs do not work for us we are moving away from docker as we build new systems, so my interest in this problem is somewhat negligible at this point.
